src/nbl/video/IPhysicalDevice.cpp

```diff
@@ -103,58 +103,226 @@ inline core::bitflag<asset::IImage::E_ASPECT_FLAGS> getImageAspects(asset::E_FOR
     uint32_t smallestTexelBlockSize = -1;
 
     auto srcChannels = asset::getFormatChannelCount(srcFormat);
-    if (!srcChannels) return asset::EF_UNKNOWN;
+    assert(srcChannels);
     auto srcAspects = getImageAspects(srcFormat);
-    auto srcBitDepth = asset::getMaxChannelBitDepth(srcFormat);
-    // TODO "Magic" value of 128.0 until I figure out a value-independent way of comparing precision
-    auto srcPrecision = getFullPrecisionValue(srcFormat, srcChannels);
+    bool srcBc = asset::isBlockCompressionFormat(srcFormat);
+
+    float srcPrecision[4];
+    float srcMinVal[4];
+    float srcMaxVal[4];
+    for (uint32_t channel = 0; channel < srcChannels; channel++)
+    {
+        srcPrecision[channel] = srcBc ? getBcFormatMaxPrecision(srcFormat, channel) : asset::getFormatPrecision(srcFormat, channel, 0.f);
+        srcMinVal[channel] = asset::getFormatMinValue<float>(srcFormat, channel);
+        srcMaxVal[channel] = asset::getFormatMaxValue<float>(srcFormat, channel);
+    }
 
     // Better way to iterate the bitflags here?
     for (uint32_t format = 0; format < asset::E_FORMAT::EF_UNKNOWN; format++)
     {
         auto f = static_cast<asset::E_FORMAT>(format);
-        if (!validFormats.hasFlags(f))
-        {
+        if (!validFormats.contains(f))
             continue;
-        }
 
+        // Can't transcode to BC or planar
+        if (asset::isBlockCompressionFormat(f))
+            continue;
+        if (asset::isPlanarFormat(f))
+            continue;
+
+        // Can't have less channels
         auto dstChannels = asset::getFormatChannelCount(f);
-        // Verify if promotion is valid from srcFormat -> format
-        bool promotionValid;
-        promotionValid = asset::isBlockCompressionFormat(f) ? f == srcFormat : true; // Can't transcode to compressed formats
-        promotionValid = promotionValid && dstChannels >= srcChannels; // Channel count
-        promotionValid = promotionValid && (getImageAspects(f) & srcAspects).value == srcAspects.value; // Aspects
-        promotionValid = promotionValid && asset::getMaxChannelBitDepth(f) >= srcBitDepth; // Data range
-        promotionValid = promotionValid && getFullPrecisionValue(f, dstChannels) >= srcPrecision; // Precision
-
-        uint32_t texelBlockSize = getTexelOrBlockBytesize(f);
-        if (promotionValid && texelBlockSize < smallestTexelBlockSize)
+        if (dstChannels < srcChannels)
+            continue;
+
+        {
+            auto c = 0;
+            while (c < srcChannels)
+            {
+                // Can't have less precision (higher precision value from getFormatPrecision) than source
+                if (asset::getFormatPrecision(f, c, 0.f) > srcPrecision[c])
+                    break;
+                // Can't have less range than source
+                if (asset::getFormatMinValue<float>(f, c) > srcMinVal[c] || asset::getFormatMaxValue<float>(f, c) < srcMaxVal[c])
```

Comment:
`<double>` otherwise you'll mess up on the full 32-bit integer format comparisons